### PR TITLE
Port remaining tests in functions.rs to sqllogictest

### LINKED ISSUE #6300: Port remaining tests in `functions.rs` to sqllogictest
### Is your feature request related to a problem or challenge?

Part of https://github.com/apache/arrow-datafusion/issues/6195. We are trying to keep the DataFusion code base clean and easy to contribute to and modify. Part of doing so is having a single location for most test coverage so new tests can be added easily and people can follow the existing patterns easily 

### Describe the solution you'd like

Port the tests in https://github.com/apache/arrow-datafusion/blob/main/datafusion/core/tests/sql/functions.rs to `scalar.slt` in https://github.com/apache/arrow-datafusion/tree/main/datafusion/core/tests/sqllogictests/test_files

Notes: 
1. Instructions for running sqllogitest are here: https://github.com/apache/arrow-datafusion/tree/main/datafusion/core/tests/sqllogictests

### Describe alternatives you've considered

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by jiangzhx**
# Which issue does this PR close?

Closes #6300.

# Rationale for this change

# What changes are included in this PR?

# Are these changes tested?

# Are there any user-facing changes?

### CODE REVIEW

File: datafusion/core/tests/sqllogictests/src/engines/conversion.rs [hunk 1]
```diff
@@ -87,5 +87,5 @@ pub fn decimal_to_str(value: Decimal) -> String {
 }
 
 pub fn big_decimal_to_str(value: BigDecimal) -> String {
-    value.round(12).normalized().to_string()
+    value.round(16).normalized().to_string()
```
Comment (jiangzhx) [inline datafusion/core/tests/sqllogictests/src/engines/conversion.rs:90]: `Sqrt(2)` returns a double value of `1.4142135623730951`. 
the precision of `value.round(12)` is not sufficient, `1.414213562373` is returned. 
To improve accuracy, we could increase the rounding precision to `value.round(16)`

File: datafusion/core/tests/sqllogictests/test_files/functions.slt [hunk 2]
```diff
@@ -479,4 +479,90 @@ SELECT struct(c1,c2,c3,c4,a,b) from simple_struct_test
...
+    arrow_cast(2, 'UInt64'),
+    arrow_cast(2, 'Float32'),
+    arrow_cast(2, 'Float64')
+)
+;
+
+query R
+SELECT sqrt(column1) FROM t
+----
+1.4142135623730951
```
Comment (jiangzhx) [inline datafusion/core/tests/sqllogictests/test_files/functions.slt:545]: The original test case had a value of `1`, but when I called `sqrt(1)` in SQLLogicTest, it returned a value of `1`, which should have been displayed as `1.0` for clarity. To make the response more reasonable, I changed the value in the test case from `1 `to `2`, it's will retrun `1.4142135623730951`

File: datafusion/core/tests/sqllogictests/test_files/functions.slt [hunk 3]
```diff
@@ -479,4 +479,90 @@ SELECT struct(c1,c2,c3,c4,a,b) from simple_struct_test
...
+SELECT "SQRT"(column1) FROM t
+
+query R
+SELECT "sqrt"(column1) FROM t
+----
+1.4142135623730951
+
+# case_builtin_math_expression
+query RRRRRRRRRR
+SELECT sqrt(column1),sqrt(column2),sqrt(column3),sqrt(column4),sqrt(column5),sqrt(column6),sqrt(column7),sqrt(column8),sqrt(column9),sqrt(column10) FROM t
```
Comment (jiangzhx) [inline datafusion/core/tests/sqllogictests/test_files/functions.slt:562]: This data could replace the original test case. Can you guys provide some feedback?

https://github.com/apache/arrow-datafusion/blob/1a1efcf7261335f1d7814be9fbb2eb455aba7e54/datafusion/core/tests/sql/functions.rs#L78-L142

File: datafusion/core/tests/sqllogictests/test_files/functions.slt [hunk 4]
```diff
@@ -479,4 +479,90 @@ SELECT struct(c1,c2,c3,c4,a,b) from simple_struct_test
...
+----
+0.6584408483418833
+
+statement ok
+drop table aggregate_test_100
+
+
+# case_sensitive_identifiers_functions
+statement ok
+create table t as values (
```
Comment (jiangzhx) [inline datafusion/core/tests/sqllogictests/test_files/functions.slt:528]: This data could replace the original test case. Can you guys provide some feedback?

https://github.com/apache/arrow-datafusion/blob/1a1efcf7261335f1d7814be9fbb2eb455aba7e54/datafusion/core/tests/sql/functions.rs#L78-L142

File: datafusion/core/tests/sqllogictests/src/engines/conversion.rs (hunk 1 above)
Comment (jiangzhx) [reply datafusion/core/tests/sqllogictests/src/engines/conversion.rs]: will cause other testcase failed,so  i reverted change.

File: datafusion/core/tests/sqllogictests/test_files/functions.slt (hunk 4 above)
Comment (alamb) [reply datafusion/core/tests/sqllogictests/test_files/functions.slt:528]: I think using 2 is a better value than 1 👍

**Review by alamb — APPROVED** [review]
I think this looks great -- thank you @jiangzhx

**Review by jackwener — APPROVED** [review]
thank you @jiangzhx

**Comment by jackwener** [comment]
I clicked by mistake, sorry ...

**Pull request merged**